From 18c843e0e6fa11f4be1f33f67c65c44405528012 Mon Sep 17 00:00:00 2001
From: Dan Kalowsky <daniel.kalowsky@intel.com>
Date: Wed, 27 Aug 2014 16:31:56 -0700
Subject: [PATCH 189/441] Altering the BYT-T-CR exit latencies

Turns out the exit latencies for BYT-T-CR are longer than standard BYT-T
by about 60 ms.  Also doubling the expecting residency time to make it
worth the transition.

These two changes provide some interesting Quadrant performance score
improvments using an IRDA-610 base:

          ECS      Anchor8
Total     +1.38%   +3.87%
CPU       -0.46%   -0.71%
Memory    +1.73%   +6.76%
IO        +0.87%   +2.37%
2D        -2.47%   +31.29%
3D        +10.18%  +11.08%

Change-Id: I4268174dbc63177fe37d507ade1cda4cf801c24b
Signed-off-by: Dan Kalowsky <daniel.kalowsky@intel.com>
(cherry picked from commit 4a56c09d6a3a3ea13a847b925dd26cbe3b7bd04e)

Signed-off-by: Nagaraj S <sivasankaranx.nagaraj@intel.com>
---
 drivers/idle/intel_idle.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -216,29 +216,29 @@ static struct cpuidle_state byt_cstates[
 		.name = "C6N-BYT",
 		.desc = "MWAIT 0x58",
 		.flags = MWAIT2flg(0x58) | CPUIDLE_FLAG_TIME_VALID | CPUIDLE_FLAG_TLB_FLUSHED,
-		.exit_latency = 40,
-		.target_residency = 275,
+		.exit_latency = 100,
+		.target_residency = 550,
 		.enter = &intel_idle },
 	{
 		.name = "C6S-BYT",
 		.desc = "MWAIT 0x52",
 		.flags = MWAIT2flg(0x52) | CPUIDLE_FLAG_TIME_VALID | CPUIDLE_FLAG_TLB_FLUSHED,
-		.exit_latency = 140,
-		.target_residency = 560,
+		.exit_latency = 200,
+		.target_residency = 1120,
 		.enter = &intel_idle },
 	{
 		.name = "C7-BYT",
 		.desc = "MWAIT 0x60",
 		.flags = MWAIT2flg(0x60) | CPUIDLE_FLAG_TIME_VALID | CPUIDLE_FLAG_TLB_FLUSHED,
-		.exit_latency = 1200,
-		.target_residency = 1500,
+		.exit_latency = 1260,
+		.target_residency = 3000,
 		.enter = &intel_idle },
 	{
 		.name = "C7S-BYT",
 		.desc = "MWAIT 0x64",
 		.flags = MWAIT2flg(0x64) | CPUIDLE_FLAG_TIME_VALID | CPUIDLE_FLAG_TLB_FLUSHED,
-		.exit_latency = 10000,
-		.target_residency = 20000,
+		.exit_latency = 10060,
+		.target_residency = 40000,
 		.enter = &intel_idle },
 	{
 		.enter = NULL }
